liblangutil/ErrorReporter.h

```diff
@@ -33,6 +33,13 @@
 namespace solidity::langutil
 {
 
+// Unique identifiers are used to tag and track individual error cases.
+// They are passed as the first parameter of error reporting functions.
+// Suffix _uid helps to find them in the sources.
+// The struct Uid prevents incidental calls like typeError(314) instead of typeError(314_uid).
+struct Uid { unsigned long long uid = 0; };
```

Comment:
Maybe `ErrorId` - we don't plan to use it for other cases that require unique IDs.

Also: Why `unsigned long long` and not just `unsigned`?

Reply:
> Maybe `ErrorId` - we don't plan to use it for other cases that require unique IDs.

I was struggling to find a short and descriptive suffix. The construct is bulky by itself, and using a long suffix makes it even more cumbersome. Also "error code" IMO is a bit misleading. We may report the same error message from different places, and then it would get different IDs.

> Also: Why `unsigned long long` and not just `unsigned`?

`unsigned` is not allowed (https://en.cppreference.com/w/cpp/language/user_literal, "Only the following parameter lists are allowed on literal operators: ...")

Reply:
What about `_error`?

Reply:
OK.